bears/general/FilenameBear.py

```diff
@@ -38,8 +46,23 @@ def run(self, filename, file,
         """
         head, tail = os.path.split(filename)
         filename_without_extension, extension = os.path.splitext(tail)
+
+        if file_naming_convention is None:
+            self.warn('Please specify a file naming convention explicitly'
+                      ' or use "auto".')
+            file_naming_convention = 'auto'
+
+        if file_naming_convention.lower() == 'auto':
+            if extension in self._language_naming_convention:
+                file_naming_convention = self._language_naming_convention[
+                    extension]
+            else:
+                self.warn('The file naming convention could not be guessed. '
+                          'Using the default "snake" naming convention...')
+                file_naming_convention = 'snake'
+
         try:
-            new_name = self._naming_convention[file_naming_convention](
+            new_name = self._naming_convention[file_naming_convention.lower()](
```

Comment:
maybe you should lower `file_naming_convention` in the beginning in L54, then you don't need to lower it for comparison and here too. (You could do for example sth like this:
```
if file_naming_convention is None:
    self.warn...
    file_naming ... = auto
else:
    file_naming_convention = file_naming_convention.lower()
```
)